sympy/combinatorics/homomorphisms.py

```diff
@@ -496,19 +500,16 @@ def group_isomorphism(G, H, isomorphism=False):
             raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
         _H, h_isomorphism = H._to_perm_group()
 
-    if not isomorphism:
-        # Prime ordered groups are cyclic.
-        from sympy import sieve
-        if (g_order in sieve) and (h_order in sieve):
-            if g_order == h_order:
-                return True
-            return False
-
     if (g_order != h_order) and (G.is_abelian != H.is_abelian):
         if not isomorphism:
             return False
         return (False, None)
 
+    if not isomorphism:
+        n = g_order
+        if (g_order == h_order) and (igcd(n, totient(n))) == 1:
+            return True
```

Comment:
Could be useful to leave a comment explaining that in this case the group is cyclic.